concourse/model/traits/release.py

```diff
@@ -515,7 +515,12 @@ def process_pipeline_args(self, pipeline_args: JobVariant):
         # validate assets if present
         for asset in self.trait.assets:
             if not pipeline_args.has_step(asset.step_name):
-                raise ValueError(f'{asset=}\'s step_name refers to an absent build-step')
+                raise ValueError(
+                    f'{asset=}\'s step_name refers to an absent build-step. If the step in '
+                    'question is declared branch-specifically, i.e. via `branch.cfg`, and the '
+                    'current branch is going to be merged with a branch declaring the pipeline '
+                    'step, this error can be safely ignored.'
```

Comment:
maybe it would be helpful to explicitly state that the error can be ignored iff the branch used for the valided pullrequest is transient only / no component-descriptor w/ related release-assets is to be released from it